gluoncv/auto/estimators/image_classification/image_classification.py

```diff
@@ -562,19 +605,30 @@ def _predict_proba(self, x, ctx_id=0):
                 batch = mx.gluon.utils.split_and_load(batch, ctx_list=self.ctx, even_split=False)
                 pred = [self.net(input) for input in batch]
                 for p in pred:
-                    probs = mx.nd.softmax(p, axis=-1)
+                    if self._problem_type in [MULTICLASS, BINARY]:
+                        probs = mx.nd.softmax(p, axis=-1)
+                    else:
+                        probs = p
                     for ii in range(p.shape[0]):
                         prob = probs[ii]
-                        results.append({'image_proba': prob.asnumpy().flatten().tolist(), 'image': x[idx]})
+                        if self._problem_type in [MULTICLASS, BINARY]:
+                            results.append({'image_proba': prob.asnumpy().flatten().tolist(), 'image': x[idx]})
+                        elif self._problem_type == REGRESSION:
```

Comment:
Raise if it's asking for the probability since there is no prob for regression problems.